lib/promscrape/discovery/puppetdb/resource.go

```diff
@@ -0,0 +1,170 @@
+package puppetdb
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape/discoveryutils"
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/promutils"
+)
+
+const (
+	separator = ","
+)
+
+var matchContentTypeRegex = regexp.MustCompile(`^(?i:application\/json(;\s*charset=("utf-8"|utf-8))?)$`)
```

Comment:
Non other discovery components perform such check. I'd suggest to simply drop it.